fili-core/src/main/java/com/yahoo/bard/webservice/web/endpoints/EndpointServlet.java

```diff
@@ -46,43 +74,124 @@ public EndpointServlet(ObjectMappersSuite objectMappers) {
-                        apiRequest.getUriInfo(),
-                        jsonName,
-                        objectMappers
-                ).getResponseStream();
-                break;
+        if (CSV.accepts(apiRequest.getFormat().toString())) {
+            builder.header(HttpHeaders.CONTENT_TYPE, "text/csv; charset=utf-8")
+                    .header(
+                            HttpHeaders.CONTENT_DISPOSITION,
+                            responseUtils.getCsvContentDispositionValue(containerRequestContext)
+                    );
+            output = new CsvResponse<>(
+                    rows,
+                    pagination,
+                    uriInfo,
+                    csvColumnNames,
+                    objectMappers
+            ).getResponseStream();
+        } else {
+            // JSON is the default
+            builder.header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON + "; charset=utf-8");
+
+            output = new JsonResponse<>(
+                    rows,
+                    pagination,
+                    containerRequestContext.getUriInfo(),
+                    jsonName,
+                    objectMappers
+            ).getResponseStream();
+
         }
         return builder.entity(output).build();
     }
+
+    /**
+     * Format and build the response as JSON or CSV.
+     *
+     * @param apiRequest  The api request object
+     * @param containerRequestContext  The context of the http request
+     * @param rows  The stream that describes the data to be formatted
+     * @param jsonName  Top-level title for the JSON data
+     * @param csvColumnNames  Header for the CSV data
+     * @param <T> The type of rows being processed
+     *
+     * @return The updated response builder with the new link header added
+     */
+    protected <T> Response formatAndPaginateResponse(
+            ApiRequest apiRequest,
+            ContainerRequestContext containerRequestContext,
+            Collection<T> rows,
+            String jsonName,
+            List<String> csvColumnNames
+    ) {
+        UriInfo uriInfo = containerRequestContext.getUriInfo();
+
+        Pagination<T> pagination;
+        pagination = new AllPagesPagination<>(
+                rows,
+                apiRequest.getPaginationParameters().orElse(apiRequest.getDefaultPagination())
+        );
+
+
+        Response.ResponseBuilder builder = responseBuilderSupplier.get();
+
+        Stream<T> stream = PageLinkBuilder.paginate(builder, pagination, uriInfo);
+        Response.ResponseBuilder responseBuilder = Response.status(Response.Status.OK);
+
+        Response response = formatResponse(
+                apiRequest,
+                responseBuilder,
+                pagination,
+                containerRequestContext,
+                stream,
+                jsonName,
+                csvColumnNames
+        );
+        return response;
+    }
+
+    /**
+     * Format and build the response as JSON or CSV.
```

Comment:
Should the JavaDoc for this method be different than the one above? It looks like it's doing more...